lib/iiif_print/split_pdfs/pages_into_images_service.rb

```diff
@@ -5,125 +5,133 @@
 
 module IiifPrint
   module SplitPdfs
+    # The purpose of this class is to split the PDF into constituent TIFF files.
+    #
+    # @see #each
     class PagesIntoImagesService
-      include Enumerable
-
-      def initialize(path)
+      DEFAULT_COMPRESSION = 'lzw'.freeze
+      def initialize(path, compression: DEFAULT_COMPRESSION, tmpdir: Dir.mktmpdir, default_dpi: 400)
         @baseid = SecureRandom.uuid
         @pdfpath = path
-        @info = nil
-        @entries = nil
-        @tmpdir = nil
-        @size = nil
-        @pagecount = nil
-        @pdftext = nil
-        @compression = 'lzw'
+        @pdfinfo = IiifPrint::SplitPdfs::PdfImageExtractionService.new(@pdfpath)
+        @tmpdir = tmpdir
+        @compression = compression
+        @default_dpi = default_dpi
       end
 
-      # return
-      def pdfinfo
-        @info = IiifPrint::SplitPdfs::PdfImageExtractionService.new(@pdfpath) if @info.nil?
-        @info
+      # In creating {#each} we get many of the methods of array operation (e.g. #to_a).
+      include Enumerable
+
+      # @api public
+      #
+      # @yieldparam [String] the path to the page's tiff.
+      def each
```

Comment:
Is this an example of monkey patching? Could you tell us more.

Reply:
The `Enumerable` module provides all kinds of methods (e.g. `#count`, `#size`, `#to_a`, etc) and assumes you've defined an `#each` method on the class.

This would not be a case of monkey patching, but instead is very much Ruby idiomatic, and helps the class behave more like an "array" without needing to inherit from Array.

Our decorator pattern is a more classic example of MonkeyPatching.